nilearn/plotting/html_stat_map.py

```diff
@@ -1,157 +1,488 @@
+        Returns: vmin, vmax
+    """
+
+    # Get vmin vmax
+    show_nan_msg = False
+    if vmax is not None and np.isnan(vmax):
+        vmax = None
+        show_nan_msg = True
+    if vmin is not None and np.isnan(vmin):
+        vmin = None
+        show_nan_msg = True
+    if show_nan_msg:
+        nan_msg = ('NaN is not permitted for the vmax and vmin arguments.\n'
+                   'Tip: Use np.nanmax() instead of np.max().')
+        warnings.warn(nan_msg)
+
+    # Extract defaults if needed
+    if vmax is None:
+        vmax = np.nanmax(data)
+    if vmin is None:
+        vmin = np.nanmin(data)
+
+    # Check that vmin and vmax make sense
+    if vmax <= vmin:
+        raise ValueError("vmax needs to be greater than vmin."
+                         " Maybe an empty image?")
+
+    return vmin, vmax
+
+
+def _threshold_data(data, threshold=None):
+    """ Threshold a data array.
+        Returns: data (masked array), threshold (updated)
+    """
+    # If threshold is None, do nothing
+    if threshold is None:
+        return data, threshold
+
+    # Deal with automatic settings of plot parameters
+    if threshold == 'auto':
+        # Threshold epsilon below a percentile value, to be sure that some
+        # voxels pass the threshold
+        threshold = fast_abs_percentile(data) - 1e-5
+
+    # Threshold
+    threshold = check_threshold(threshold, data,
+                                percentile_func=fast_abs_percentile,
+                                name='threshold')
+
+    # Mask data
+    if threshold == 0:
+        data = np.ma.masked_equal(data, 0, copy=False)
+    else:
+        data = np.ma.masked_inside(data, -threshold, threshold, copy=False)
+    return data, threshold
+
+
+def _save_sprite(data, output_sprite, vmax, vmin, mask=None, cmap='Greys',
+                 format='png'):
+    """ Generate a sprite from a 3D Niimg-like object.
+        Returns: sprite
+    """
+
+    # Create sprite
+    sprite = _data_to_sprite(data)
+
+    # Mask the sprite
+    if mask is not None:
+        mask = _data_to_sprite(mask)
+        sprite = np.ma.array(sprite, mask=mask)
+
+    # Save the sprite
+    imsave(output_sprite, sprite, vmin=vmin, vmax=vmax, cmap=cmap,
+           format=format)
+
+    return sprite
+
+
+def _bytesIO_to_base64(handle_io):
+    """ Encode the content of a bytesIO virtual file as base64.
+        Also closes the file.
+        Returns: data
+    """
+    handle_io.seek(0)
+    data = encodebytes(handle_io.read()).decode('utf-8')
+    handle_io.close()
+    return data
+
+
+def _save_cm(output_cmap, cmap, format='png', n_colors=256):
+    """ Save the colormap of an image as an image file.
+    """
+
+    # save the colormap
+    data = np.arange(0., n_colors) / (n_colors - 1.)
+    data = data.reshape([1, n_colors])
+    imsave(output_cmap, data, cmap=cmap, format=format)
+
+
+def _deduplicate_cmap(cmap, annotate, n_colors=256):
+    """Make sure that the colormap has no duplicated colors.
```

Comment:
I have added a line to the doc to clarify:
`If a color is used more than once, it will be removed from the color map.`

The colormap is then reinterpolated to the specified number of colors.